--- a/home-manager/home-manager
+++ b/home-manager/home-manager
@@ -634,7 +634,22 @@
     presentNews
 }
 
+function doUpdate() {
+    setFlakeAttribute
+    pathInputs=$(nix eval --impure --expr "
+        let
+          inherit (import <nixpkgs> {}) lib;
+          pathInputs = lib.filterAttrs (n: v: lib.hasPrefix \"path:\" v.url) (import (lib.removePrefix \"path:\" $FLAKE_ARG/flake.nix )).inputs;
+          pathInputNames = builtins.attrNames pathInputs;
+        in
+        toString pathInputNames
+    " | tr -d '"')
+    nix flake update --flake "$FLAKE_ARG" "$pathInputs"
+}
+
 function doSwitch() {
+    doUpdate
+
     setWorkDir
 
     local generation
